src/main/java/com/google/api/codegen/config/FieldConfig.java

```diff
@@ -212,10 +212,14 @@ public FieldConfig withResourceNameConfig(ResourceNameConfig resourceNameConfig)
   public boolean requiresParamTransformation() {
     return getResourceNameConfig() != null
         && getMessageResourceNameConfig() != null
-        && getMessageResourceNameConfig().getResourceNameType() != ResourceNameType.ANY
         && !getResourceNameConfig().equals(getMessageResourceNameConfig());
   }
 
+  public boolean requiresAnyParamTransformation() {
```

Comment:
I think that does make sense. But `TestAssertionTransformation` isn't quite right, because we also do the transformation in a non-test context when creating the oneof object (but NOT when converting to string for the ANY case).

Perhaps because the new way of handling the ANY case has introduced a difference between the test case and the non-test case, these should be handled separately, meaning that the logic should be pulled out of `FieldConfig` completely, and implemented where it is used in the `InitCodeTransformer` (for test assertion creation) and in the `StaticLangApiMethodTransformer` for method body generation.